sample-compose/app/src/main/java/co/nimblehq/sample/compose/di/modules/PreferencesModule.kt

```diff
@@ -6,27 +6,26 @@ import androidx.datastore.preferences.core.Preferences
 import androidx.datastore.preferences.preferencesDataStore
 import co.nimblehq.sample.compose.data.repository.AppPreferencesRepositoryImpl
 import co.nimblehq.sample.compose.domain.repository.AppPreferencesRepository
-import dagger.Module
-import dagger.Provides
+import dagger.*
 import dagger.hilt.InstallIn
 import dagger.hilt.android.qualifiers.ApplicationContext
 import dagger.hilt.components.SingletonComponent
-import javax.inject.Singleton
-
-private val Context.appPreferencesDataStore by preferencesDataStore("app_preferences")
 
 @Module
 @InstallIn(SingletonComponent::class)
-class PreferencesModule {
+abstract class PreferencesModule {
+
+    @Binds
+    abstract fun bindAppPreferencesRepository(
+        pppPreferencesRepositoryImpl: AppPreferencesRepositoryImpl
```

Comment:
Here and everywhere below: 😆 
```suggestion
        appPreferencesRepositoryImpl: AppPreferencesRepositoryImpl
```